flink-staging/flink-streaming/flink-streaming-core/src/test/java/org/apache/flink/streaming/api/functions/PrintSinkFunctionTest.java

```diff
@@ -0,0 +1,332 @@
+			return 0;
+		}
+
+		@Override
+		public int getIndexInSubtaskGroup() {
+			return 0;
+		}
+
+		@Override
+		public InputSplitProvider getInputSplitProvider() {
+			return null;
+		}
+
+		@Override
+		public IOManager getIOManager() {
+			return null;
+		}
+
+		@Override
+		public MemoryManager getMemoryManager() {
+			return null;
+		}
+
+		@Override
+		public String getTaskName() {
+			return null;
+		}
+
+		@Override
+		public String getTaskNameWithSubtasks() {
+			return null;
+		}
+
+		@Override
+		public ClassLoader getUserClassLoader() {
+			return null;
+		}
+
+		@Override
+		public Map<String, Future<Path>> getDistributedCacheEntries() {
+			return new Map<String, Future<Path>>() {
+				@Override
+				public int size() {
+					return 0;
+				}
+
+				@Override
+				public boolean isEmpty() {
+					return false;
+				}
+
+				@Override
+				public boolean containsKey(Object key) {
+					return false;
+				}
+
+				@Override
+				public boolean containsValue(Object value) {
+					return false;
+				}
+
+				@Override
+				public Future<Path> get(Object key) {
+					return null;
+				}
+
+				@Override
+				public Future<Path> put(String key, Future<Path> value) {
+					return null;
+				}
+
+				@Override
+				public Future<Path> remove(Object key) {
+					return null;
+				}
+
+				@Override
+				public void putAll(Map<? extends String, ? extends Future<Path>> m) {
+
+				}
+
+				@Override
+				public void clear() {
+
+				}
+
+				@Override
+				public Set<String> keySet() {
+					return null;
+				}
+
+				@Override
+				public Collection<Future<Path>> values() {
+					return null;
+				}
+
+				@Override
+				public Set<Entry<String, Future<Path>>> entrySet() {
+					return null;
+				}
+			};
+		}
+
+		@Override
+		public BroadcastVariableManager getBroadcastVariableManager() {
+			return null;
+		}
+
+		@Override
+		public AccumulatorRegistry getAccumulatorRegistry() {
+			return null;
+		}
+
+		@Override
+		public void acknowledgeCheckpoint(long checkpointId) {
+
+		}
+
+		@Override
+		public void acknowledgeCheckpoint(long checkpointId, StateHandle<?> state) {
+
+		}
+
+		@Override
+		public ResultPartitionWriter getWriter(int index) {
+			return null;
+		}
+
+		@Override
+		public ResultPartitionWriter[] getAllWriters() {
+			return new ResultPartitionWriter[0];
+		}
+
+		@Override
+		public InputGate getInputGate(int index) {
+			return null;
+		}
+
+		@Override
+		public InputGate[] getAllInputGates() {
+			return new InputGate[0];
+		}
+	};
+
+	public OutputStream out = new OutputStream() {
+		@Override
+		public void write(int b) throws IOException {
+
+		}
+	};
+
+	@Test
+	public void testPrintSinkStdOut(){
+
+		printStreamMock stream = new printStreamMock(out);
+		System.setOut(stream);
+
+		ExecutionConfig executionConfig = new ExecutionConfig();
+		final HashMap<String, Accumulator<?, ?>> accumulators = new HashMap<String, Accumulator<?, ?>>();
+		final StreamingRuntimeContext ctx = new StreamingRuntimeContext(
```

Comment:
You can replace all this by `final StreamingRuntimeContext ctx = Mockito.mock(StreamingRuntimeContext.class)`.